### DISCUSSION TITLE
Bigger OLED and Choc pockets

### DISCUSSION THREAD

**Discussion by freznel10** [joshreve/dactyl-keyboard#19] [Ideas]
Hi, I was wondering the OLEDs could be made to support bigger oleds (128x64)? Will the HS pockets fit Choc hotswap sockets as well?

**Comment by joshreve** [comment]
The code for the OLED will allow a larger one, you just need to change the dimension in the config for the OLED and the left wall.  If you plan to build I can help with which dimensions you need to change.  I'm hesitant to create even more specific configs for OLED, but it might be helpful to create some example snippets of code for different examples.

Let me know if you want to build for a larger OLED.  I can tell you what parameters to modify.

**Reply by joshreve** [reply]
Chocs would only need someone to create a FreeCAD file for the plate.  I can convert it for import, but I don't have a geometry to export from for Chocs.  If you know of any already open sourced I can incorporate it easily.  I may post a poll on that and if it were popular enough I can make the geometry myself.  I'm trying to focus my time where there is the most interest.

**Comment by freznel10** [comment]
Hi Josh,

Thanks for the revert. In looking at using the .96 128 X 64 oleds.

**Reply by joshreve** [reply]
Here are the parameters I would start with, comments in CAPS:

DEFINITELY NEED TO CHANGE, MEASURE WIDTH AND ADD A LITTLE GAP
'oled_mount_width': 12.5,  # whole OLED width

PROBABLY WANT TO MAKE LARGER BY ABOUT THE SAME AMOUNT AS MOUNT WIDTH
'oled_left_wall_x_offset_override': 24.0,

 MAY BE THE SAME
'oled_mount_height': 39.0,  # whole OLED length

PROBABLY NEED TO SHIFT AROUND A BIT DEPENDING ON CONNECTOR PINS IN BACK AND SCREEN LOCATION ON FRONT
'oled_mount_connector_hole': 6.0,
'oled_screen_start_from_conn_end': 6.5,
'oled_screen_length': 24.5,
'oled_screen_width': 10.5,


The generator creates test files for the OLED mount so take a guess and print one to test.  Shouldn't be too bad.  If you are willing to send the values when you are done, I'd like to at least add notes to the file for the next person trying it.

**Comment by freznel10** [comment]
Ohh. Ill search if there is something we could us

**Comment by freznel10** [comment]
Awesome. I will do test prints and report back. Thank you so much.

**Comment by freznel10** [comment]
First test print:


Will need to adjust height more.

**Comment by freznel10** [comment]
Here's the second test print with the following settings:

'oled_mount_width': 27, # whole OLED width


'oled_left_wall_x_offset_override': 38.5,


'oled_mount_height': 28, # whole OLED length


'oled_mount_connector_hole': 5,
'oled_screen_start_from_conn_end': 5,
'oled_screen_length': 15.5,
'oled_screen_width': 25

These work well.

**Reply by Looooopy** [reply]
Do those values correspond to an 'UNDERCUT' OLED?

**Comment by freznel10** [comment]
These use OLED clips.

**Reply by joshreve** [reply]
I'll have to think about how to incorporate.  I really need to add a builds/gallery section at some point.  I think your settings will help others, but I'm torn between providing a place to find the settings and creating a switch between the standard and wide OLED.  

Do you have a choc geometry or are you holding off on that pending the OLED completion?

**Reply by Looooopy** [reply]
[Disclaimer]
I have not studied the code at all because I just started to look into how to build my own keyboard and I am not so familiar with Python code and has not generated 3d models with scripts before, so I am quite novice in this.

[But here I'll go anyway]
Why not have something like sensible defaults (For example how its setup now) and you have some kind of template system that that overrides the sensible defaults, so you can break it out into smaller pieces and make it easier to understand and test.

It looks like you have components for more complex shapes so why not break out both the part and configuration into a configurable unit  that are not included in the sensible defaults.

So if you wish to have an OLED_X (OLED_{BRAND/GENERIC}_128x32) or TRACKBALL_X (TRACKBALL_{BRAND/GENERIC}) you append and override it.

I think it would be easier for the users and in the long run it would be easier to do a web generator that people can just select their hardware and click generate and if a new hardware coming into market you have to know more about the underlying structure how its generated to be able to fiddle more.

People with deeper knowledge can do pull request for the small component that they would like to add, say for example a drawer for ESP32 or some other hardware they like to add. 

That's my thoughts about it, but I could be totally off base here - maybe its only we who thinkers that will use it?

FORGOT TO SAY: You guys have done an outstanding job of bring this to the community!

**Reply by joshreve** [reply]
The web generator is really the end game.  I have built a few flask web UIs so I will eventually get to it.  I don't have the bandwidth to tackle that for a while, but I think a docker based web generator is where we will eventually get with this.  

In the mean time, I'm just trying to keep it reasonably usable.  I think a better prebuilt gallery is really next so those without programming can at least pick from some tried and functioning models.

**Reply by Looooopy** [reply]
Nice that you already have thought about it, I think it's possible to put up something in AWS for free at least for a year, but to get a nice DNS name it would cost something.

Sure a prebuilt gallery is one way to go I can agree would satisfy most users. But also it does not hurt to do something simpler for selecting additional features but it sounds like you are working on it in your mind.
